src/ImageSharp.Web/ICachedImage.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.IO;
+
+namespace SixLabors.ImageSharp.Web
+{
+    /// <summary>
+    /// Specifies the contract for managing image buffers from different locations.
+    /// </summary>
+    public interface ICachedImage
```

Comment:
alternative name ideas:
`IImageStreamSource` or simply `IImageSource`

Reply:
I think `IImageResolver` should become `IImageSource` and `ICachedImage` should become `IImageResolver`